### PR TITLE
net/packet, wgengine/filter: support SCTP

### PR DESCRIPTION
**Pull request by bradfitz**
Server-side ACL work remains.

Updates #1516

### CODE REVIEW

File: wgengine/filter/filter.go [hunk 1]
```diff
@@ -352,7 +352,7 @@ func (f *Filter) runIn4(q *packet.Parsed) (r Response, why string) {
 		if f.matches4.match(q) {
 			return Accept, "tcp ok"
 		}
-	case packet.UDP:
+	case packet.UDP, packet.SCTP:
```
Comment (danderson) [inline wgengine/filter/filter.go:355]: Use a different LRU for each protocol, otherwise an outbound SCTP packet would allow UDP return.

**Review by danderson — APPROVED** [review]

File: wgengine/filter/filter.go (hunk 1 above)
Comment (danderson) [reply wgengine/filter/filter.go:355]: (or add the proto to the flowtrack tuple - 5-tuple of proto,srcip,dstip,srcport,dstport is a common network thing, we should have one)

Comment (bradfitz) [reply wgengine/filter/filter.go:356]: I went with the 5-tuple before I saw your follow-up comment, so I feel like that's a good answer. :)

It ended up making things nicer in other places too, like TSMP rejected header's Flow method no longer loses that info when returning the flowtrack.Tuple.

**Comment by bradfitz** [comment]
PTAL.

I had to split out ipproto into its own package for cycle reasons.

**Comment by bradfitz** [comment]
If the ipproto package & type renames are too invasive in this I could also probably split it off into its own commit.

File: net/ipproto/ipproto.go [hunk 2]
```diff
@@ -1,28 +1,32 @@
-// Copyright (c) 2020 Tailscale Inc & AUTHORS All rights reserved.
```
Comment (danderson) [inline net/ipproto/ipproto.go:1]: Put this in types/ipproto rather than net?

**Review by danderson — APPROVED** [review]

File: net/ipproto/ipproto.go (hunk 2 above)
Comment (bradfitz) [reply net/ipproto/ipproto.go]: Done.

**Pull request merged**